data/src/main/java/smile/data/Attribute.java

```diff
@@ -63,6 +63,16 @@
      */
     private double weight;
 
+    /**
+     * Value between [-1, 1] that represents monotonic regression coefficient.
```

Comment:
Is it necessary to add this variable? It is algorithm dependent if a attribute is used for monotonic regression. It is sufficient to define this in the algorithm API, rather than here.

Reply:
Yeah, I see what you mean. I think I will change it that way. 

This Attribute is used for things other than regression trees too. And monotonic regression makes sense only for regression tree model. So it's probably better to have it in Algorithm API